helix-term/src/application.rs

```diff
@@ -228,6 +229,63 @@ impl Application {
+                        let token = match params.token {
+                            lsp::NumberOrString::Number(n) => n.to_string(),
+                            lsp::NumberOrString::String(s) => s,
+                        };
+                        let msg = {
+                            let lsp::ProgressParamsValue::WorkDone(work) = params.value;
+                            match work {
+                                lsp::WorkDoneProgress::Begin(lsp::WorkDoneProgressBegin {
+                                    title,
+                                    message,
+                                    percentage,
+                                    ..
+                                }) => {
+                                    let mut status = String::new();
+                                    if let Some(percentage) = percentage {
+                                        status.push_str(percentage.to_string().as_str());
+                                        status.push_str("% ");
+                                    }
+                                    status.push_str(&title);
+                                    if let Some(message) = message {
+                                        status.push_str(" - ");
+                                        status.push_str(&message);
+                                    }
+                                    status
+                                }
+                                lsp::WorkDoneProgress::Report(lsp::WorkDoneProgressReport {
+                                    message,
+                                    percentage,
+                                    ..
+                                }) => {
+                                    let mut status = String::new();
+                                    if let Some(percentage) = percentage {
+                                        status.push_str(percentage.to_string().as_str());
+                                        status.push_str("% ");
+                                    }
+                                    if let Some(message) = message {
+                                        status.push_str(&message);
+                                    }
+                                    status
+                                }
+                                lsp::WorkDoneProgress::End(lsp::WorkDoneProgressEnd {
+                                    message,
+                                }) => {
+                                    if let Some(message) = message {
+                                        message
+                                    } else {
+                                        self.editor.clear_status();
+                                        return;
+                                    }
+                                }
+                            }
+                        };
+                        let status = format!("[{}] {}", token, msg);
```

Comment:
I think we can just show only the percentage at the second bar. Not sure if it will be better, although less verbose but we won't miss any error message.

Reply:
In my opinion if this information is available it should be displayed. Showing only percentage will, in case of rust, hide how many modules have already been indexed, so I think it's still valuable to display it.